src/main/scala/tech/cryptonomic/conseil/tezos/TezosPlatformDiscoveryOperations.scala

```diff
@@ -167,47 +203,27 @@ class TezosPlatformDiscoveryOperations(
     result.unsafeToFuture()
   }
 
-  /** Method querying slick metadata tables for entities */
-  private def preCacheEntities: DBIO[EntitiesCache] = {
-    for {
-      tables <- MTable.getTables(Some(""), Some("public"), Some(""), Some(Seq("TABLE")))
-      counts <- getTablesCount(tables)
-    } yield now -> (tables.map(_.name.name) zip counts).map {
-      case (name, count) =>
-        Entity(name, makeDisplayName(name), count)
-    }.toList
-  }
-
-  /** Query for counting rows in the table */
-  private def getTablesCount(tables: Vector[MTable]): DBIO[Vector[Int]] = {
-    DBIOAction.sequence {
-      tables.map { table =>
-        TezosDatabaseOperations.countRows(table.name.name)
-      }
-    }
-  }
-
-  /** Makes displayName out of name */
-  private def makeDisplayName(name: String): String = {
-    name.capitalize.replace("_", " ")
-  }
-
-  /** Returns current time in milliseconds */
-  private def now: Long = System.currentTimeMillis()
 
   /** Makes list of possible string values of the attributes
     *
     * @param  tableName  name of the table from which we extract attributes
-    * @param  attribute  name of the attribute
+    * @param  column  name of the attribute
     * @param  withFilter optional parameter which can filter attributes
-    * @return list of attributes
-    * */
-  def listAttributeValues(tableName: String, attribute: String, withFilter: Option[String] = None): Future[List[String]] = {
-    for {
-      attrOpt <- attributesCache.read.map(_.get(tableName)
-        .flatMap(_._2.find(_.name == attribute))).unsafeToFuture()
-      result <- attrOpt.map(attr => makeAttributesQuery(tableName, attr.name, withFilter)).toList.sequence
-    } yield result.flatten
+    * @return Either list of attributes or list of errors
+    **/
+  def listAttributeValues(tableName: String, column: String, withFilter: Option[String] = None): Future[Either[List[AttributesValidationError], List[String]]] = {
+    val result = for {
+      attrOpt <- getTableAttributes(tableName).map(_.flatMap(_.find(_.name == column)))
+    } yield {
+      val invalidDataTypeValidationResult = if (!attrOpt.exists(attr => canQueryType(attr.dataType))) List(InvalidAttributeDataType(column)) else Nil
```

Comment:
Wouldn't it be easier to create a list of Options and then flatten it to get the `res`?

Reply:
I'm not sure I follow, can you elaborate on you idea?

Reply:
```scala
val invalidDataTypeFailure: Option[Invalid...] = ???
val highCardinalityFailure: Option[HighCard....] = ???

val res = List(invalid..., high...).flatten
```

Reply:
ok, changed